File: Irrgarten/lib/UI/TextUI.rb
require 'io/console'
require_relative '../Directions'

module UI

  class TextUI

    #https://gist.github.com/acook/4190379
    def read_char
      STDIN.echo = false
      STDIN.raw!
    
      input = STDIN.getc.chr
      if input == "\e" 
        input << STDIN.read_nonblock(3) rescue nil
        input << STDIN.read_nonblock(2) rescue nil
      end
    ensure
      STDIN.echo = true
      STDIN.cooked!
    
      return input
    end

    def next_move
      print "Where? "
      got_input = false
      while (!got_input)
        c = read_char
        case c
          when "\e[A"
            puts "UP ARROW"
            output = Irrgarten::Directions::UP
            got_input = true
          when "\e[B"
            puts "DOWN ARROW"
            output = Irrgarten::Directions::DOWN
            got_input = true
          when "\e[C"
            puts "RIGHT ARROW"
            output = Irrgarten::Directions::RIGHT
            got_input = true
          when "\e[D"
            puts "LEFT ARROW"
            output = Irrgarten::Directions::LEFT
            got_input = true
          when "\u0003"
            puts "CONTROL-C"
            got_input = true
            exit(1)
          else
            #Error
        end
      end
      output
    end

    def show_game(game_state)
		puts "Jugadores:\n"+game_state.getPlayers()
		puts "Jugadores:\n"+game_state.getPlayers()
        puts "Monstruos:\n"+game_state.getMonsters()
        puts "Jugador actual:"+ (game_state.getCurrentPlayer() + 1)
        puts "\nEstado del laberinto\n"+game_state.getLabyrinth()
        puts "\nWinner:"+game_state.getWinner()
        if(!game_state.getWinner())
            puts "\nLog:"+game_state.getLog()
        end
    end

  end # class   

end # module   

File: irrgarten/src/UI/TextUI.java

package UI;

/**
 *
 * @author Mangel
 */
import irrgarten.Directions;
import irrgarten.GameState;
import java.util.Scanner;


public class TextUI {
    
    private static Scanner in = new Scanner(System.in);
    
    private char readChar() {
        String s = in.nextLine();     
        return s.charAt(0);
    }
    

    public Directions nextMove() {
        System.out.print("Where? ");
        
        Directions direction = Directions.DOWN;
        boolean gotInput = false;
        
        while (!gotInput) {
            char c = readChar();
            switch(c) {
                case 'w':
                    System.out.print(" UP\n");
                    direction = Directions.UP;
                    gotInput = true;
                    break;
                case 's':
                    System.out.print(" DOWN\n");
                    direction = Directions.DOWN;
                    gotInput = true;
                    break;
                case 'd':
                    System.out.print("RIGHT\n");
                    direction = Directions.RIGHT;
                    gotInput = true;
                    break;
                case 'a':
                    System.out.print(" LEFT\n");
                    direction = Directions.LEFT;
                    gotInput = true;    
                    break;
            }
        }    
        return direction;
    }
    
    public void showGame(GameState gameState) {   
        System.out.print("Jugadores:\n"+gameState.getPlayers());
        System.out.print("Monstruos:\n"+gameState.getMonsters());
        System.out.print("Jugador actual:"+ (gameState.getCurrentPlayer() + 1));
        System.out.print("\nEstado del laberinto\n"+gameState.getLabyrinth());
        System.out.print("\nWinner:"+gameState.getWinner());
        if(!gameState.getWinner())
            System.out.print("\nLog:"+gameState.getLog());
    }
    
}